sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
+ that give users access to the LRO and expose methods to interact with them such as waiting until the operation reaches a terminal state, checking its status, or
+ providing a callback to do work on the final result when it is ready. If the LRO follows the [Azure REST API guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#long-running-operations--jobs), 
+ it's likely that the generated client library code should _just work_. 
+ In cases where the LRO diverges from the guidelines, you may need to customize your code to achieve the desired scenario.
+
+ There are 3 options to customize the logic for our LROs.
+
+ 1) [LRO algorithm/strategies - OperationResourcePolling, LocationPolling, StatusCheckPolling](#lro-algorithmstrategies---operationresourcepolling-locationpolling-statuscheckpolling)
+    - You need to customize the polling strategy
+ 2) [Polling method - LROBasePolling/AsyncLROBasePolling](#polling-method---lrobasepollingasynclrobasepolling)
+    - You need to customize the polling loop
+ 3) [Poller - LROPoller/AsyncLROPoller](#poller---lropollerasynclropoller)
+    - You need to customize the public interface of the poller
+
+
+### LRO algorithm/strategies - OperationResourcePolling, LocationPolling, StatusCheckPolling
+
+`azure.core.polling` provides three built-in strategies for polling - `OperationResourcePolling`, `LocationPolling`, and `StatusCheckPolling`.
+The type of polling needed will be determined automatically, unless otherwise changed by the client library developer. If the LRO is determined
+not to fit `OperationResourcePolling` or `LocationPolling`, `StatusCheckPolling` serves as a fallback strategy which will not perform polling, but
+instead return a successful response for a 2xx.
+
+If you need to customize the polling strategy, choose a polling algorithm that closely represents what you need to do
+or create your own that inherits from `azure.core.polling.LongRunningOperation` and implements the necessary methods.
+
+For our purposes, let's say that `OperationResourcePolling` closely resembles what the service does, but we
+need to account for a non-standard status.
+
+Example: Raise exception for a non-standard status that is returned via the initial POST call - "ValidationFailed".
+
+```python
+class CustomOperationResourcePolling(OperationResourcePolling):
+    """Implements a operation resource polling, typically from Operation-Location. Customized to raise an
+    exception in the case of a "ValidationFailed" status returned from the service.
+
+    :param str operation_location_header: Name of the header to return operation format (default 'operation-location')
+    :keyword dict[str, any] lro_options: Additional options for LRO. For more information, see
+     https://aka.ms/azsdk/autorest/openapi/lro-options
```

Comment:
Not sure if we need these params in the docstring, given that we don't overwrite the constructor?